src/Microsoft.Graph.Core/Requests/GraphClientFactory.cs

```diff
@@ -0,0 +1,259 @@
+        /// <see cref="HttpRequestMessage"/> travels from the <see cref="HttpClient"/> to the network and an 
+        /// <see cref="HttpResponseMessage"/> travels from the network back to <see cref="HttpClient"/>.
+        /// The handlers are invoked in a top-down fashion. That is, the first entry is invoked first for 
+        /// an outbound request message but last for an inbound response message.</param>
+        /// <returns>An <see cref="HttpClient"/> instance with the configured handlers.</returns>
+        public static HttpClient CreateClient(params DelegatingHandler[] handlers)
+        {
+            return CreateClient(defaultTimeout, _baseAddress, null, null, handlers);
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="HttpClient"/> instance configured with the handlers provided.
+        /// </summary>
+        /// <param name="innerHandler">The inner handler represents the destination of the HTTP message channel.</param>
+        /// <param name="handlers">An ordered list of <see cref="DelegatingHandler"/> instances to be invoked as an 
+        /// <see cref="HttpRequestMessage"/> travels from the <see cref="HttpClient"/> to the network and an 
+        /// <see cref="HttpResponseMessage"/> travels from the network back to <see cref="HttpClient"/>.
+        /// The handlers are invoked in a top-down fashion. That is, the first entry is invoked first for 
+        /// an outbound request message but last for an inbound response message.</param>
+        /// <returns>An <see cref="HttpClient"/> instance with the configured handlers.</returns>
+        public static HttpClient CreateClient(HttpMessageHandler innerHandler, params DelegatingHandler[] handlers)
+        {
+            return Create(innerHandler, handlers);
+        }
+        /// <summary>
+        /// Creates a new <see cref="HttpClient"/> instance configured with the handlers provided.
+        /// </summary>
+        /// <param name="sovereignCloud">The <see cref="GraphServieCloudList"/> enum value to pass the national cloud root endpoint to client.</param>
+        /// <param name="handlers">An ordered list of <see cref="DelegatingHandler"/> instances to be invoked as an 
+        /// <see cref="HttpRequestMessage"/> travels from the <see cref="HttpClient"/> to the network and an 
+        /// <see cref="HttpResponseMessage"/> travels from the network back to <see cref="HttpClient"/>.
+        /// The handlers are invoked in a top-down fashion. That is, the first entry is invoked first for 
+        /// an outbound request message but last for an inbound response message.</param>
+        /// <returns>An <see cref="HttpClient"/> instance with the configured handlers.</returns>
+        public static HttpClient CreateClient(GraphServieCloudList sovereignCloud, params DelegatingHandler[] handlers)
+        {
+            string cloud = cloudList[(int)sovereignCloud];
+            string cloudAddress = cloud + "/" + version;
```

Comment:
We should expose version as an option when creating a Graph client.

Reply:
Yes. I'd like to set version to be V1.0 default and expose the Set method for it.